lib/pbench/test/unit/server/test_api_key.py

```diff
@@ -90,4 +92,162 @@ def test_successful_api_key_generation(self, query_get_as, pbench_drb_token):
+        audit = Audit.query()
+        assert len(audit) == 2
+        assert audit[0].id == 1
+        assert audit[0].root_id is None
+        assert audit[0].operation == OperationCode.DELETE
+        assert audit[0].status == AuditStatus.BEGIN
+        assert audit[0].name == "apikey"
+        assert audit[0].object_type == AuditType.API_KEY
+        assert audit[0].object_id is None
+        assert audit[0].object_name is None
+        assert audit[0].user_id == "3"
+        assert audit[0].user_name == "drb"
+        assert audit[0].reason is None
+        assert audit[0].attributes is None
+        assert audit[1].id == 2
+        assert audit[1].root_id == 1
+        assert audit[1].operation == OperationCode.DELETE
+        assert audit[1].status == AuditStatus.SUCCESS
+        assert audit[1].name == "apikey"
+        assert audit[1].object_type == AuditType.API_KEY
+        assert audit[1].object_id is None
+        assert audit[1].object_name is None
+        assert audit[1].user_id == "3"
+        assert audit[1].user_name == "drb"
+        assert audit[1].reason is None
+        assert audit[1].attributes["key_id"] == pbench_drb_api_key.id
+        assert audit[1].attributes["name"] == "drb_key"
+
+        assert APIKey.query(api_key=pbench_drb_api_key.api_key) is None
+
+    def test_unauthorized_delete(
+        self, query_as, pbench_drb_token_invalid, pbench_drb_api_key
+    ):
+
+        response = query_as(
+            pbench_drb_token_invalid,
+            pbench_drb_api_key.id,
+            HTTPStatus.UNAUTHORIZED,
+        )
+        assert response.json == {
+            "message": "User provided access_token is invalid or expired"
+        }
+
+    def test_delete_api_key_notfound(self, query_as, pbench_drb_token):
+        response = query_as(
+            pbench_drb_token, "pbench_drb_token_invalid", HTTPStatus.NOT_FOUND
+        )
+        assert response.json == {"message": "Requested key not found"}
+        audit = Audit.query()
+        assert len(audit) == 2
+        assert audit[0].id == 1
+        assert audit[0].root_id is None
+        assert audit[0].operation == OperationCode.DELETE
+        assert audit[0].status == AuditStatus.BEGIN
+        assert audit[0].name == "apikey"
+        assert audit[0].object_type == AuditType.API_KEY
+        assert audit[1].id == 2
+        assert audit[1].root_id == 1
+        assert audit[1].operation == OperationCode.DELETE
+        assert audit[1].status == AuditStatus.FAILURE
+        assert audit[1].name == "apikey"
+        assert audit[1].object_type == AuditType.API_KEY
+        assert audit[1].attributes is None
+
+    def test_delete_api_key_fail(
+        self, query_as, pbench_admin_token, pbench_drb_api_key
+    ):
+
+        response = query_as(
+            pbench_admin_token, pbench_drb_api_key.id, HTTPStatus.FORBIDDEN
+        )
+        assert response.json == {
+            "message": "User does not have rights to delete the specified key"
+        }
```

Comment:
Yeah, I think it will be an important feature in terms of addressing security breaches:  if a user's API key is leaked/stolen, having to wait for _that user_ to remedy the situation is a potential problem for the Server deployment; an Admin user should be able to revoke another user's key.